clients/spark/core/src/test/resources/parser-test/sst_files_generator.go

```diff
@@ -57,11 +90,15 @@ func newGenerateNanoid(size int) func() (string, error) {
 }
 
 func newGenerateFuzz() func() (string, error) {
+	return newGenerateFuzzWithSharedPrefixes(EmptyPrefix)
+}
+
+func newGenerateFuzzWithSharedPrefixes(sharedPrefix string) func() (string, error) {
 	f := fuzz.NewWithSeed(FuzzerSeed)
 	return func() (string, error) {
-		var val string
-		f.Fuzz(&val)
-		return val, nil
+		var suffix string
+		f.Fuzz(&suffix)
+		return sharedPrefix + suffix, nil
```

Comment:
Can we vary the length of the shared prefix between adjacent elements?  E.g. maybe along these lines:

* Start with an empty prefix, sharedPrefixLen = 1 (say)
* Loop:
  * New key = prefix + random(10 bytes)
  * Randomly perform one of these actions (say each with probability 1/4, but it doesn't really matter):
    - Keep sharedPrefixLen.
    - Decrement sharedPrefixLen if it's >0
    - Increment sharedPrefixLen if it's <10
    - Set sharedPrefixLen = 0
  * Set prefix = substring(key, 0, sharedPrefixLen)

Now sharedPrefixLen will fluctuate between entries, but it _will_ change it it **will** occasionally become 0.  Might be a bit noisier.

WDYT?

Reply:
I agree with the idea but took another approach to achieve this goal that worked better with the implementation. let me know if this addresses your comment.